### DISCUSSION TITLE
Controller redirectToRoute uses wrong port with symfony server

### DISCUSSION THREAD

**Discussion by michnovka** [symfony/symfony#47690] [Q&A]
### Symfony version(s) affected

6.1.4

### Description

When using local symfony server with custom port, redirects are made to the default 8000 port:

```
symfony server:start --port=8001
```

### How to reproduce

In my controller I call

```php
return $this->redirectToRoute('cp_dashboard');
```

which is supposed to redirect to my dashboard controller:

```php
class DashboardController extends AbstractCPController
{
    #[Route('/', name: 'cp_dashboard')]
    public function index(): Response
    {
        return $this->render('dashboard/index.html.twig');
    }
}
```

The response HTTP 302 location header leads to port 8000 instead of the one in use (8001). 

![image](https://user-images.githubusercontent.com/16553087/192060238-e97e9dbe-2eb9-4383-98bd-2421174348b1.png)
![image](https://user-images.githubusercontent.com/16553087/192060284-a3b0aec4-d3d4-4caa-8179-411806b1d83d.png)

here are my routes.yaml

```yaml
controllers:
    resource: ../src/Controller/
    type: attribute
    prefix: /cp
```

I do not have port 8000 hardcoded anywhere in my code (I did search in all files)

**Comment by monteiro** [comment]
Hi @michnovka, 

I have tried to recreate your bug but I couldn't. Can you help me with this sample project to mimic it? If we can mimic it we can fix it.

https://github.com/monteiro/symfony-47677

**Comment by michnovka** [comment]
The issue seems to be caused when I have more projects running.

I have one project on 127.0.0.1:8000. This project has controller for POST /cp/login and it returns 302 /cp/invoices/show

Now when I post to the same path, but on the other project (127.0.0.1:8001/cp/login) the controller of the 2nd project does handle the request, but the reply has 302 redirect to the :8000.

I went over profiler and see what I found:

![image](https://user-images.githubusercontent.com/16553087/192120520-f6cdf60d-1624-4e31-a0b9-c2b15cbb0323.png)

Do you see the

```
_security.main.target_path | "https://127.0.0.1:8000/cp/billing/invoice/1/details/show"
```

How did it end up on the :8001 server which has nothing to do with the :8000 project. As if they were sharing the same session somehow

**Reply by stof** [reply]
cookies are shared between different ports on the same domain. So if you don't customize the session cookie name, you will share the session cookie in such setup.
